pkg/consensus/client.go

```diff
@@ -94,27 +103,39 @@ func (c *Client) loadCurrentContext(ctx context.Context, currentSlot types.Slot,
 	return nil
 }
 
+func (c *Client) GetProposerCache(ctx context.Context, slot types.Slot) (ValidatorInfo, bool) {
+	val, ok := c.proposerCache.Get(slot)
+	if !ok {
+		return ValidatorInfo{}, ok
+	}
+	validator, ok := val.(ValidatorInfo)
+	return validator, ok
+}
```

Comment:
I think we should be fine to communicate the absence with an error instead of the boolean `ok` value, and alongside we can just return a pointer to the value so we can get the standard `return nil, err` text for the error case

something like this:

```suggestion
	val, ok := c.proposerCache.Get(slot)
	if !ok {
		return nil, fmt.Errorf("could not find proposer for slot %d", slot)
	}
	validator, ok := val.(ValidatorInfo)
	if !ok {
        	return nil, fmt.Errorf("internal: proposer cache contains an unexpected type %T", val) 
        }
	return &validator, nil
```